oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClientAuthenticationMethod.java

```diff
@@ -31,6 +31,12 @@
 	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
 	public static final ClientAuthenticationMethod BASIC = new ClientAuthenticationMethod("basic");
 	public static final ClientAuthenticationMethod POST = new ClientAuthenticationMethod("post");
+
+	/**
```

Comment:
Add javadoc

Reply:
BASIC and POST don't have anything. Should I just differentiate this by saying something like "@Since 5.2, For public clients that don't require client authentication."?